manager/orchestrator/task.go

```diff
@@ -65,3 +75,119 @@ func IsTaskDirty(s *api.Service, t *api.Task) bool {
 	return !reflect.DeepEqual(s.Spec.Task, t.Spec) ||
 		(t.Endpoint != nil && !reflect.DeepEqual(s.Spec.Endpoint, t.Endpoint.Spec))
 }
+
+// InvalidNode is true if the node is nil, down, or Drain
+func InvalidNode(n *api.Node) bool {
+	return n == nil ||
+		n.Status.State == api.NodeStatus_DOWN ||
+		n.Spec.Availability == api.NodeAvailabilityDrain
+}
+
+type isRelatedService func(*api.Service) bool
+
+// InitTasks fix tasks in store before orchestrator run. Previous manager might haven't finished processing their updates.
```

Comment:
`InitTasks fixes tasks in the store before the orchestrator runs. The previous leader might not have finished processing their updates and left them in an inconsistent state.`